### PR TITLE
[ci-skip] Wording in guides: Prefer environment over mode

### PR DESCRIPTION
**Pull request by robinboening**
In some guides the word `mode` refers to environment and in some its using the word environment. Since the word `mode` is used for many different things it might be confusing for many people.

@eugeneius pointed this out in his comment https://github.com/rails/rails/pull/40309#discussion_r497366580, so I picked it up and went through all the guides and changed the wording.

### CODE REVIEW

**Review by jeremy — COMMENTED** [review]
Agree with dropping "mode."

I'd expect to read "in development" or "in the development environment" rather than "in development environment," however.

**Comment by robinboening** [comment]
You're right. I already had a feeling it reads a bit weird, but as a non-native speaker I could tell if my feeling was right or not. I'll fix it tomorrow.

**Comment by robinboening** [comment]
@jeremy could you check again? Sometimes I used "in development" and sometimes "in the development environment". I often omitted "environment" if it was already used in the same or in the previous sentence.

File: guides/source/action_view_overview.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ more examples and information.
 
 #### Template Caching
 
-By default, Rails will compile each template to a method in order to render it. When you alter a template, Rails will check the file's modification time and recompile it in development mode.
+By default, Rails will compile each template to a method in order to render it. In the development environment, when you alter a template, Rails will check the file's modification time and recompile it.
```
Comment (jeremy) [inline guides/source/action_view_overview.md:184]: Out of scope for this PR, but worth expanding this section as well:

```suggestion
In production, template caching is enabled. Rails optimizes for app speed. Template files are loaded once and compiled into Ruby methods for fast rendering.

In development, template caching is disabled. Rails optimizes for rapid developer feedback. Rails detects changes you make to template files (by checking file modification time) and automatically recompiles them so you can see your changes immediately.

Template caching is governed by `config.action_view.cache_template_loading` which defaults to the value of `config.cache_classes`: true in the production and test environments and false in development.
```

**Review by jeremy — APPROVED** [review]

**Pull request merged**